src/python/pants/backend/python/util_rules/dists.py

```diff
@@ -203,14 +207,14 @@ async def run_pep517_build(request: DistBuildRequest, python_setup: PythonSetup)
         ),
     )
 
-    dist_dir = "dist"
+    output_path: str = str(dist_dir.relpath / request.output_path)
```

Comment:
OK, I've just realized that I was wrong, and this `"dist"` does need to be hardcoded, because this is not Pants's dist dir, but setuptools's dist directory, which we control, and leave as the default `dist/`. So what we actually want to do is revert the change I recommended  here (sorry) and still strip dist down below, but then *add* the output_path as a new prefix, with AddPrefix.

Reply:
But what we should add here is a comment explaining this, so that people in the future don't make the same mistake I did...

Reply:
Something like - "# This is the setuptools dist directory, not Pants's, so we harcode to dist/."

Reply:
Basically we invoke setuptools inside the sandbox, so we strip the dist/ prefix it adds and then later we add our own.

Reply:
Okay, I tried to make everything work correctly with the hard-coded `"dist"` but I'm not sure that what I did is correct. I also added the comment you requested.

Reply:
I'm not sure whether I have now is correct, but it seems that the files in `result` were already within `os.path.join(dist_dir, request.output_path)`. There was no need to add the `output_path` prefix, so I just removed the `dist_dir` prefix as before.

Reply:
I think what you have is correct. But you should test it out manually to see that it does what you expect.

Reply:
`output_path` seems to work the way I want in manual testing.